src/Microsoft.Azure.SignalR/HubHost/ServiceLifetimeManagerBase.cs

```diff
@@ -18,15 +19,35 @@ internal abstract class ServiceLifetimeManagerBase<THub> : HubLifetimeManager<TH
         protected const string NullOrEmptyStringErrorMessage = "Argument cannot be null or empty.";
         protected const string TtlOutOfRangeErrorMessage = "Ttl cannot be less than 0.";
         protected readonly IServiceConnectionManager<THub> ServiceConnectionContainer;
+        protected readonly IClientInvocationManager ClientInvocationManager;
+        protected readonly IClientConnectionManager ClientConnectionManager;
         protected ILogger Logger { get; set; }
 
         private readonly DefaultHubMessageSerializer _messageSerializer;
-
-        public ServiceLifetimeManagerBase(IServiceConnectionManager<THub> serviceConnectionManager, IHubProtocolResolver protocolResolver, IOptions<HubOptions> globalHubOptions, IOptions<HubOptions<THub>> hubOptions, ILogger logger)
+        private readonly string _callerId;  
+
+        public ServiceLifetimeManagerBase(
+            IServiceConnectionManager<THub> serviceConnectionManager,
+            IHubProtocolResolver protocolResolver,
+            IOptions<HubOptions> globalHubOptions,
+            IOptions<HubOptions<THub>> hubOptions,
+            IServerNameProvider nameProvider,
+            IClientInvocationManager clientInvocationManager,
+            IClientConnectionManager clientConnectionManager,
+            ILogger logger)
         {
             Logger = logger ?? throw new ArgumentNullException(nameof(logger));
             ServiceConnectionContainer = serviceConnectionManager;
             _messageSerializer = new DefaultHubMessageSerializer(protocolResolver, globalHubOptions.Value.SupportedProtocols, hubOptions.Value.SupportedProtocols);
+
+            if (nameProvider == null)
+            {
+                throw new ArgumentNullException(nameof(nameProvider));
+            }
+            _callerId = nameProvider.GetName();
+
+            ClientInvocationManager = clientInvocationManager ?? throw new ArgumentNullException(nameof(clientInvocationManager));
+            ClientConnectionManager = clientConnectionManager ?? throw new ArgumentNullException(nameof(clientConnectionManager));
```

Comment:
ClientConnectionManager does not make sense for Management SDK, so I don't think we should move it to base

Reply:
moved to `ServiceLifetimeManagerBase`